lib/app/dispatcher.rb

```diff
@@ -56,46 +57,74 @@ def pool
-
-          # connector-level checks
-          unless Connectors::REGISTRY.registered?(service_type)
-            Utility::Logger.info("The service type (#{service_type}) of connector (ID: #{connector_id}) is not supported. Skipping...")
-            next
-          end
-          if index_name.nil? || index_name.empty?
-            Utility::Logger.info("The index name of connector (ID: #{connector_id}, service type: #{service_type}) is empty. Skipping...")
-            next
-          end
-          unless index_name.start_with?(Utility::Constants::CONTENT_INDEX_PREFIX)
-            Utility::Logger.info("The index name of connector (ID: #{connector_id}, service type: #{service_type}) is invalid, it must start with '#{Utility::Constants::CONTENT_INDEX_PREFIX}'. Skipping...")
-            next
-          end
-          Core::ElasticConnectorActions.ensure_content_index_exists(index_name)
-
-          pool.post do
-            send_heartbeat(connector_id, service_type)
-            Utility::Logger.info("Starting a job for connector (ID: #{connector_id}, service type: #{service_type})...")
-            job_runner = Core::SyncJobRunner.new(connector_settings, service_type)
-            job_runner.execute
-          rescue StandardError => e
-            Utility::ExceptionTracking.log_exception(e, "Job for connector (ID: #{connector_id}, service type: #{service_type}) failed due to unexpected error.")
+        scheduler.when_triggered do |connector_settings, task|
+          case task
+          when :sync
+            start_sync_task(connector_settings)
+          when :heartbeat
+            start_heartbeat_task(connector_settings)
+          when :configuration
+            start_configuration_task(connector_settings)
+          else
+            Utility::Logger.error("Unknown task type: #{task}. Skipping...")
           end
         end
       rescue StandardError => e
-        Utility::ExceptionTracking.log_exception(e, 'connector service failed due to unexpected error.')
+        Utility::ExceptionTracking.log_exception(e, 'The connector service failed due to unexpected error.')
       end
 
-      def send_heartbeat(connector_id, service_type)
-        Utility::Logger.info("Sending a heartbeat on [#{connector_id} - #{service_type}]...")
-        Core::Heartbeat.send(connector_id, service_type)
+      def start_sync_task(connector_settings)
+        # connector-level checks
+        unless Connectors::REGISTRY.registered?(connector_settings.service_type)
+          Utility::Logger.info("The service type (#{connector_settings.service_type}) is not supported. Skipping...")
+          return
+        end
+        unless connector_settings.valid_index_name?
+          Utility::Logger.info("The index name of #{connector_settings.formatted} is invalid. Skipping...")
+          return
+        end
+        Core::ElasticConnectorActions.ensure_content_index_exists(connector_settings.index_name)
+
+        start_heartbeat_task(connector_settings)
```

Comment:
Hmmm, so one task calls another? Is it needed here?

Reply:
Yes, according to [the connector protocol](https://github.com/elastic/connectors-ruby/blob/main/docs/CONNECTOR_PROTOCOL.md):

>Connectors should update the last_seen field with their current datetime in UTC format (timezone-agnostic) every half hour, and every time they perform a sync job

Reply:
I've seen it in 2 places, that's why I'm asking:
![image](https://user-images.githubusercontent.com/12238374/190976826-5f9cf285-de40-4867-8b21-0f29e600ba74.png)

I was curious if it's needed in both, I would remove it from the sync task if possible

Reply:
Indeed, `start_heartbeat_task` is called in 2 places, and this is according to the protocol, that heartbeat will be sent 
1. every half an hour
2. every time a sync job runs